src/main/java/io/cdap/plugin/sap/odata/odata2/OData2Client.java

```diff
@@ -0,0 +1,165 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.sap.odata.odata2;
+
+import io.cdap.plugin.sap.odata.EntityType;
+import io.cdap.plugin.sap.odata.ODataClient;
+import io.cdap.plugin.sap.odata.ODataEntity;
+import io.cdap.plugin.sap.odata.PropertyMetadata;
+import io.cdap.plugin.sap.odata.exception.ODataException;
+import org.apache.olingo.odata2.api.edm.Edm;
+import org.apache.olingo.odata2.api.edm.EdmEntitySet;
+import org.apache.olingo.odata2.api.edm.EdmEntityType;
+import org.apache.olingo.odata2.api.edm.EdmException;
+import org.apache.olingo.odata2.api.edm.EdmProperty;
+import org.apache.olingo.odata2.api.ep.EntityProvider;
+import org.apache.olingo.odata2.api.ep.EntityProviderException;
+import org.apache.olingo.odata2.api.ep.EntityProviderReadProperties;
+import org.apache.olingo.odata2.api.ep.feed.ODataFeed;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Provides handy methods to consume OData v2 services using Apache Olingo v2 as Client Library.
+ */
+public class OData2Client extends ODataClient {
+
+  /**
+   * Default {@link EntityProvider} settings to read an entity.
+   */
+  private static final EntityProviderReadProperties READ_PROPERTIES = EntityProviderReadProperties.init().build();
+
+  /**
+   * Cache metadata to avoid repetitive API calls.
+   */
+  private Edm metadata;
+
+  /**
+   * @param rootUrl  URL of the OData service. The URL must end with an external service name
+   *                 (e.g., http://eccsvrname:8000/sap/opu/odata/sap/zgw100_dd02l_so_srv/).
+   * @param username username for basic authentication.
+   * @param password password for basic authentication.
+   */
+  public OData2Client(String rootUrl, String username, String password) {
+    super(rootUrl, username, password);
+  }
+
+  /**
+   * Get OData service metadata.
+   *
+   * @return OData service metadata.
+   * @throws ODataException if the metadata cannot be fetched.
+   */
+  public Edm getMetadata() {
+    if (metadata == null) {
+      initMetadata();
+    }
+    return metadata;
+  }
+
+  @Override
+  public Iterator<ODataEntity> queryEntitySet(String entitySetName, @Nullable String query) {
+    Edm metadata = getMetadata();
+    URI queryURI = getQueryURI(entitySetName, query);
+    HttpURLConnection connection = connect(queryURI.toASCIIString(), MediaType.APPLICATION_ATOM_XML);
+    try {
+      EdmEntitySet entitySet = metadata.getDefaultEntityContainer().getEntitySet(entitySetName);
+      InputStream content = (InputStream) connection.getContent();
```

Comment:
Normally an InputStream is closed by whoever creates it. 

Is this closed by EntityProvider.readFeed()? If so, please add a comment. If not, please be sure to close it.

Reply:
Added `content.close();` method invokation.